server/src/main/java/org/opensearch/telemetry/metrics/WrappedMetricsRegistry.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.metrics;
+
+import org.opensearch.common.annotation.InternalApi;
+import org.opensearch.telemetry.TelemetrySettings;
+
+import java.io.IOException;
+
+/**
+ * Wrapper implementation of {@link MetricsRegistry}. This delegates call to right {@link MetricsRegistry} based on the settings
+ */
+@InternalApi
+public class WrappedMetricsRegistry implements MetricsRegistry {
+
+    private final MetricsRegistry defaultMetricsRegistry;
+    private final TelemetrySettings telemetrySettings;
+
+    /**
+     * Constructor.
+     * @param defaultMetricsRegistry default meter registry.
+     * @param telemetrySettings telemetry settings.
+     */
+    public WrappedMetricsRegistry(TelemetrySettings telemetrySettings, MetricsRegistry defaultMetricsRegistry) {
+        this.telemetrySettings = telemetrySettings;
+        this.defaultMetricsRegistry = defaultMetricsRegistry;
+    }
+
+    @Override
+    public Counter createCounter(String name, String description, String unit) {
+        return new WrappedCounter(telemetrySettings, defaultMetricsRegistry.createCounter(name, description, unit));
```

Comment:
Only difference between tracer::startSpan and Meter is the life span. Meters mostly will be created once and that too at class instantiation and will be incremented/updated on each execution. So disabling metrics framework should be able to stop reporting the metrics for all the meters. Your thoughts on this?

Reply:
>  So disabling metrics framework should be able to stop reporting the metrics for all the meters. Your thoughts on this?

Ideally, we should just close the `TracingTelemetry` and `MetricsTelemetry`: even being "disabled", those are not - we still have all reporters etc sitting in background and doing nothing. For metrics it is even more weird since:
a) we still keep old counter around
b) flipping metrics on / off would report weird values (with the gaps between tuning off or on)

Without wrapping, we at least keep the counter accurate.

Reply:
>>Ideally, we should just close the TracingTelemetry and MetricsTelemetry

We can close these by passing the callback to TelemetrySettings from the Node.java but enabling it back would be challenging. We can keep this way also that only disabling once is allowed through dynamic settings and to enable it back requires a cluster restart.

Reply:
@reta Should we try closing the metricsTelemetry on disable? Also one more thing may be will address in the other PR but should we also add a Setting to enable/disable the telemetry feature fe somebody just want to enable the Metrics but not Tracing or vice versa

Reply:
@Gaganjuneja I think our mechanism with enabling / disabling is not up to the promise with how it is currently being implemented (using wrappers), in terms of OTEL, we have following implementation options:
 - "disable tracing" means set sampler to "never sample"
 - "disable metrics" means use Drop aggregation (https://opentelemetry.io/docs/specs/otel/metrics/sdk/#drop-aggregation)

It would make sense to reconsider current approach, otherwise we would end up with ever growing amount of wrappers and noop implementations;

> @reta Should we try closing the metricsTelemetry on disable?

So if we think about both options, nor closing or disabling (as we did for tracing) would be acceptable, because with either a) end up with meters having very weird values b) or mix of real and noop meters. The best option I see is to:
 - use metrics provider as is
 - use Drop aggregation when metrics are disabled (so they won't be recorded but will be counted)